esp-hal/src/dma/mod.rs

```diff
@@ -414,15 +434,15 @@ macro_rules! dma_circular_descriptors {
 #[macro_export]
 macro_rules! dma_buffers_chunk_size {
```

Comment:
Fun sneaky breaking change: these now return a `&mut [u8]` instead of `&mut [u8; T/RX_SIZE]`, which is funny because it can't be directly passed to mem2mem start_transaction.

Reply:
> Unfortunately, PDMA writes out of bounds for transfers of length 2 and 3

Is this in the TRM or esp-idf?

Reply:
IDF just downright ignores the problem and aligns the buffers by default to either 4 bytes or to some cache-dependent value that I ignored.

I didn't find anything in the 32/S2 TRMs, maybe I missed it.

However, I did find RX bytes in my TX buffer during testing :) Before the RX/TX swap we were simply lucky that the overflow was written to somewhere we didn't care about in the tests.

Reply:
The ESP32 TRM says buffer and size of the descriptor must be word aligned for all peripherals.

The ESP32-S2 TRM seems to allow byte-addressable access. Did you observe the out of bounds access on the S2 as well?

Mind adding a comment somewhere that the word alignment is only required for the ESP32 (and possibly S2 if confirmed)?